src/customEvent.shim.js

```diff
@@ -7,48 +9,46 @@ if (typeof PST_NO_SHIM === 'undefined') {
   var HashChangeEvent = root.HashChangeEvent;
   var Event = root.Event;
 
-  (function () {
-    if (!Element.prototype.addEventListener) {
-      return;
-    }
-
-    function CustomEvent(event, params) {
-      params = params || {
-        bubbles: false,
-        cancelable: false,
-        detail: undefined
-      };
-      var evt = document.createEvent('CustomEvent');
-      evt.initCustomEvent(event, params.bubbles, params.cancelable, params.detail);
-      return evt;
-    }
-
-    CustomEvent.prototype = Event.prototype;
-
-    if (!root.CustomEvent || isIE()) {
-      root.CustomEvent = CustomEvent;
-    }
-
-    // Opera before 15 has HashChangeEvent but throw a DOM Implement error
-    if (!HashChangeEvent || (root.opera && root.opera.version() < 15) || isIE()) {
-      HashChangeEvent = root.CustomEvent;
-    }
-
-    if (isIE()) {
-      Event = CustomEvent;
-    }
-
-    // fix for Safari
-    try {
-      new HashChangeEvent('hashchange');
-    } catch (e) {
-      HashChangeEvent = CustomEvent;
-    }
-
-    try {
-      new Event('popstate');
-    } catch (e) {
-      Event = CustomEvent;
-    }
-  })();
+  if (!Element.prototype.addEventListener) {
+    throw new Error('CustomEvent shim needs EventTarget support to work.');
+  }
+
+  let CustomEvent = function CustomEvent(event, params) {
+    params = params || {
+      bubbles: false,
+      cancelable: false,
+      detail: undefined
+    };
```

Comment:
all or nothing is ok here, because by default they can be `undefined` that is same as `false`, so if define only one of the params, it will be valid usage :)